common/static/common/js/discussion/views/discussion_inline_view.js

```diff
@@ -0,0 +1,231 @@
+
+            window.user = user;
+            DiscussionUtil.setUser(user);
+            Content.loadContentInfos(response.annotated_content_info);
+            DiscussionUtil.loadRoles(response.roles);
+
+            this.course_settings = new DiscussionCourseSettings(response.course_settings);
+
+            this.discussion = new Discussion(undefined, {pages: response.num_pages});
+            this.discussion.reset(response.discussion_data, {
+                silent: false
+            });
+
+            discussionHtml = edx.HtmlUtils.template($('#inline-discussion-template').html())({
+                threads: response.discussion_data,
+                read_only: this.readOnly,
+                discussionId: discussionId
+            });
+
+            if (this.$('section.discussion').length) {
+                edx.HtmlUtils.setHtml(this.$el, discussionHtml);
+                this.$('section.discussion').replaceWith(edx.HtmlUtils.ensureHtml(discussionHtml).toString());
+            } else {
+                edx.HtmlUtils.append(this.$el, discussionHtml);
+            }
+
+            this.threadListView = new DiscussionThreadListView({
+                el: this.$('.inline-threads'),
+                collection: self.discussion,
+                courseSettings: self.course_settings
+            });
+
+            this.threadListView.render();
+
+            this.threadListView.on('thread:selected', _.bind(this.navigateToThread, this));
+
+            DiscussionUtil.bulkUpdateContentInfo(window.$$annotated_content_info);
+
+            this.newPostForm = this.$el.find('.new-post-article');
+
+            this.newPostView = new NewPostView({
+                el: this.newPostForm,
+                collection: this.discussion,
+                course_settings: this.course_settings,
+                topicId: discussionId,
+                is_commentable_cohorted: response.is_commentable_cohorted
+            });
+
+            this.newPostView.render();
+
+            this.listenTo(this.newPostView, 'newPost:createPost', this.hideNewPost);
+            this.listenTo(this.newPostView, 'newPost:cancel', this.hideNewPost);
+            this.discussion.on('add', this.addThread);
+
+            this.retrieved = true;
+            this.showed = true;
+
+            if (this.isWaitingOnNewPost) {
+                this.newPostForm.removeClass('is-hidden').focus();
+            }
+
+            // Hide the thread view initially
+            this.$('.inline-thread').addClass('is-hidden');
+        },
+
+        navigateToThread: function(threadId) {
+            var thread = this.discussion.get(threadId);
+            this.threadView = new DiscussionThreadView({
+                el: this.$('.forum-content'),
+                model: thread,
+                mode: 'tab',
+                course_settings: this.course_settings
+            });
+            this.threadView.render();
+            this.threadListView.$el.addClass('is-hidden');
```

Comment:
@andy-armstrong during your last commit, the line @bjacobel added is lost, but appears in my PR. 

this.listenTo(this.threadView.showView, 'thread:_delete', this.navigateToAllPosts);

https://github.com/edx/edx-platform/pull/14076/files#diff-22e5bd3fa33b76d64644ef0e07813172R148